truffleruby/src/main/java/org/truffleruby/cext/CExtNodes.java

```diff
@@ -1085,4 +1088,38 @@ protected ObjectIVarSetNode createObjectIVarSetNode() {
 
     }
 
+    @CoreMethod(names = "capture_exception", onSingleton = true, needsBlock = true)
+    public abstract static class StoreExceptionNode extends YieldingCoreMethodNode {
+
+        @Specialization
+        public Object executeWithProtect(VirtualFrame frame, DynamicObject block,
+                @Cached("create()") BranchProfile exceptionProfile) {
+            try {
+                yield(frame, block);
```

Comment:
Also add a BranchProfile here, so if it always throws the compiler knows we never return `nil()` and only throw exceptions.